nixos/doc/manual/development/option-types.section.md

```diff
@@ -197,14 +197,14 @@ merging is handled.
 
 Submodules are detailed in [Submodule](#section-option-types-submodule).
 
-`types.submodule` *`o`*
-
-:   A set of sub options *`o`*. *`o`* can be an attribute set, a function
-    returning an attribute set, a list of attribute sets, or a path to a
-    file containing such a value. When the argument is a list, it is
-    equivalent to the argument being a set with the argument as the value
-    of the `imports` attribute. Submodules are used in composed types to
-    create modular options. This is equivalent to
+`types.submodule` *`module`*
+
+:   A nested module evaluation, containing its own set of options.
+    `module` can either be a single module (in the form of an attribute
+    set, a function, or a path), or as a list of modules (which then
+    gets processed just like a list assigned to `imports`). Submodules can be
+    used in composed types to create modular options. This type is
+    equivalent to
```

Comment:
The equivalence is an intrinsic property of submodules whereas the preceding sentence is about usage. Could you flip those around?

Reply:
Do you mean to first write that it is the same as `submoduleWith ...` and then describe the usage?